starport/services/network/join.go

```diff
@@ -0,0 +1,218 @@
+		return 0, false, err
+	}
+	b.ev.Send(events.New(events.StatusDone, "AddValidator transaction sent"))
+
+	return requestRes.RequestID, requestRes.AutoApproved, nil
+}
+
+// CreateAccountRequestMsg creates an add AddAccount request message
+func (b *Builder) CreateAccountRequestMsg(
+	ctx context.Context,
+	chainHome string,
+	customGentx bool,
+	launchID uint64,
+	amount sdk.Coin,
+) (uint64, bool, bool, error) {
+	address := b.account.Address(SPNAddressPrefix)
+	spnAddress, err := SetSPNPrefix(address)
+	if err != nil {
+		return 0, false, false, err
+	}
+
+	b.ev.Send(events.New(events.StatusOngoing, "Verifying account already exists "+spnAddress))
+
+	shouldCreateAcc := false
+	if !customGentx {
+		exist, err := CheckGenesisAddress(chainHome, spnAddress)
+		if err != nil {
+			return 0, false, false, err
+		}
+		if !exist {
+			exist, err = b.CheckAccountExist(ctx, launchID, spnAddress)
+			if err != nil {
+				return 0, false, false, err
+			}
+		}
+		shouldCreateAcc = !exist
+	}
+	if shouldCreateAcc || customGentx {
+		msg := launchtypes.NewMsgRequestAddAccount(
+			spnAddress,
+			launchID,
+			sdk.NewCoins(amount),
+		)
+
+		b.ev.Send(events.New(events.StatusOngoing, "Broadcasting account transactions"))
+		res, err := b.cosmos.BroadcastTx(b.account.Name, msg)
+		if err != nil {
+			return 0, false, false, err
+		}
+
+		var requestRes launchtypes.MsgRequestResponse
+		if err := res.Decode(&requestRes); err != nil {
+			return 0, false, false, err
+		}
+		b.ev.Send(events.New(events.StatusDone, "AddAccount transactions sent"))
+
+		return requestRes.RequestID, requestRes.AutoApproved, true, nil
+	}
+
+	b.ev.Send(events.New(events.StatusDone, "Account already exist"))
+	return 0, false, false, nil
+}
+
+// GetAccountAddress return an account address for the blockchain by name
+func (b *Blockchain) GetAccountAddress(ctx context.Context, accountName string) (string, error) {
+	if !b.isInitialized {
+		return "", errors.New("the blockchain must be initialized to show an account")
+	}
+
+	chainCmd, err := b.chain.Commands(ctx)
+	if err != nil {
+		return "", err
+	}
+	acc, err := chainCmd.ShowAccount(ctx, accountName)
+	if err != nil {
+		return "", err
+	}
+	return acc.Address, nil
+}
+
+// CheckAccountExist check if the account already exists or is pending approval
+func (b *Builder) CheckAccountExist(ctx context.Context, launchID uint64, address string) (bool, error) {
+	if b.hasAccount(ctx, launchID, address) {
+		return true, nil
+	}
+	// verify if the account is pending approval
+	requests, err := b.fetchRequests(ctx, launchID)
+	if err != nil {
+		return false, err
+	}
+	for _, request := range requests {
+		switch req := request.Content.Content.(type) {
+		case *launchtypes.RequestContent_GenesisAccount:
+			if req.GenesisAccount.Address == address {
+				return true, nil
+			}
+		case *launchtypes.RequestContent_VestingAccount:
+			if req.VestingAccount.Address == address {
+				return true, nil
+			}
+		}
+	}
```

Comment:
Why? this way, we can avoid someone flooding the requests sending a lot of requests

Reply:
They can flood requests without using Starport CLI if they want anyway. This would allows the user to directly send a new request if they want to correct the account balance for example without sending a reject request
Personal opinion open for debate

Reply:
hmm, true, make sense!! the user can send different amounts and self-delegation. changed